### PR TITLE
set internal field in specific variables in the counterexample trace for test-gen-support

### PR DESCRIPTION
**Pull request by lucasccordeiro**
hide variable assignments related to dynamic_object[0-9], dynamic_[0-9]_array, and CPROVER internal functions (e.g., __CPROVER_initialize). This PR fixes diffblue/test-gen#20.

### CODE REVIEW

File: src/goto-symex/build_goto_trace.cpp [hunk 1]
```diff
@@ -132,6 +132,79 @@ exprt adjust_lhs_object(
 
 /*******************************************************************\
 
+Function: hide_dynamic_object
```
Comment (peterschrammel) [inline src/goto-symex/build_goto_trace.cpp]: does not match function name

File: src/goto-symex/build_goto_trace.cpp [hunk 2]
```diff
@@ -132,6 +132,79 @@ exprt adjust_lhs_object(
...
+
+  Inputs:
+
+ Outputs:
+
+ Purpose: hide variable assignments related to dynamic_object[0-9]
```
Comment (peterschrammel) [inline src/goto-symex/build_goto_trace.cpp]: is "hide" still correct?

File: src/goto-symex/build_goto_trace.cpp [hunk 3]
```diff
@@ -132,6 +132,79 @@ exprt adjust_lhs_object(
...
+  else
+  {
+    forall_operands(it, expr)
+      set_internal_dynamic_object(*it, goto_trace_step, ns);
+  }
+}
+
+/*******************************************************************\
+
+Function: update_fields_to_hidden
```
Comment (peterschrammel) [inline src/goto-symex/build_goto_trace.cpp]: is "hidden" still correct?

File: src/goto-symex/build_goto_trace.cpp (hunk 1 above)
Comment (lucasccordeiro) [inline src/goto-symex/build_goto_trace.cpp]: Implemented as suggested.

File: src/goto-symex/build_goto_trace.cpp (hunk 2 above)
Comment (lucasccordeiro) [inline src/goto-symex/build_goto_trace.cpp]: Implemented as suggested.

File: src/goto-symex/build_goto_trace.cpp (hunk 3 above)
Comment (lucasccordeiro) [inline src/goto-symex/build_goto_trace.cpp]: Updated method signature.

**Pull request merged**